### DISCUSSION TITLE
Get early access to New Fulfillment Inbound API v2024-03-20

### DISCUSSION THREAD

**Discussion by jevoniuk** [amzn/selling-partner-api-models#3833] [Announcements]
We are excited to announce early access to the new Fulfillment Inbound API v2024-03-20. These APIs support FBA inbound shipment creation and will allow you to:  

- Select from multiple inbound placement service options  
- Access and edit a shipment interchangeably on both Send to Amazon and via APIs  
- Create larger plans, with support for 1500 SKUs/plan, instead of 200 SKUs/plan on APIs today  
- Retrieve shipment information (e.g., Amazon reference number, shipment status)

With this launch, we are announcing the **deprecation of 13 operations within Fulfillment Inbound API v0 which will be removed on December 20, 2024**. If you use Fulfillment Inbound API v0, please plan your migration and begin integrating with this new version using the [Fulfillment Inbound v2024-03-20 API use case guide](https://developer-docs.amazon.com/sp-api/docs/fulfillment-inbound-api-v2024-03-20-use-case-guide). For more information and a list of deprecated operations refer to the [Fulfillment Inbound API FAQs](https://developer-docs.amazon.com/sp-api/docs/fulfillment-inbound-faq).

Early access provides you with preview of functionality that may not be fully supported until the APIs are generally available. For the Fulfillment Inbound API v2024-03-20, additional functionality will become available after the APIs are generally available, including: 

- **New feature:** Viewing and selecting one of the available delivery windows for a shipment to arrive at an Amazon facility. This is a change from the functionality described in the use case guide, which allows a client to submit a delivery window start date when confirming a transportation option for a shipment (refer to [`confirmTransportationOptions`](https://developer-docs.amazon.com/sp-api/docs/fulfillment-inbound-api-v2024-03-20-reference#confirmtransportationoptions)).  
- **New feature:** Printing item labels  
- Selecting Partnered Carrier as an option for LTL shipments  
- Updating the name of inbound plans and shipments  
- Updating the quantity of units of a SKU after shipment confirmation

## Which stores are affected?

The ability to create inbound shipments with the new Fulfillment Inbound API v2024-03-20 is available in the following stores:  

- Canada
- US  
- Spain  
- UK  
- France  
- Netherlands  
- Germany  
- Italy  
- Sweden  
- Singapore  
- Australia  
- Japan 

We will announce support for the following stores as they become available:  

- Mexico  
- Brazil  
- Poland  
- Egypt  
- Turkey  
- Saudi Arabia  
- United Arab Emirates  
- India

**Comment by git-mike-z** [comment]
The new API is missing GET shipments. Any workaround?

**Reply by ben-pierre** [reply]
You have to get inbound plans, loop over the inbound plans to pull shipment ids, then call get shipment on each one from there. With the new shipment splits we're doing ~40 shipments FTL/LTL a week. That combined with the fact each operation takes 10 seconds means getting a list of shipments for the week via API takes several minutes. Incredible.

**Reply by git-mike-z** [reply]
And then, you have to call list Inbound Plan Items.
So:

1. List active inbound plans
2. get plan ids & shipment ids
3. get shipment
4. get inbound plan items

Previously, after getting a list of shipments we get items etc with each shipment read. Half the steps

**Reply by ben-pierre** [reply]
Worth noting that last inbound plan items does not match items to shipments, just to plans. It is not currently possible to get items on a shipment via the new API, and the old one is being deprecated in December. 

New API is good to go and totally ready to ship though, nbd 👎

**Comment by BorisR63** [comment]
Previously with getShipmentItems it was possible to receive QuantityReceived data that shown the item quantity that has been received at an Amazon fulfillment center.
Is there any way to get this quantity with the new API ?

**Comment by Ironkyy** [comment]
Fulfillment Inbound API v2024-03-20 Why is the Amazon partnered carrier selected by default?How to use Fulfillment Inbound API v2024-03-20 to modify the carrier to a non-Amazon partner carrier

**Comment by ericblade** [comment]
Isn't the point of versioned APIs that you don't need to deprecate them? 

That said, reviewing the docs, I can't figure out which one is *more* difficult.  THey are both inscrutable.

**Comment by tamt** [comment]
try to listInboundPlans of my Germany store, i got:
```
{
    "errors": [
        {
            "code": "Unauthorized",
            "message": "Access to requested resource is denied.",
            "details": ""
        }
    ]
}
```

**Reply by SuperFlyingCow** [reply]
I get Unauthorized access when trying to use the latest Inbound API. Any suggestion?

**Comment by git-mike-z** [comment]
Make sure you refresh your access token.

**Reply by SuperFlyingCow** [reply]
Hi, already did. Do I need to wait for a couple hours for it to take effect? I can access Fulfillment Inbound API v0 just fine but not this latest API.

**Comment by git-mike-z** [comment]
Should be immediate. I can flip between the APIs without issue. Maybe
double check your target url? IDepending on region you're accessing, the
base URI changes, but the rest is :
{{baseUrl}}/inbound/fba/2024-03-20/inboundPlans

On Thu, Jul 18, 2024 at 9:29 AM SuperFlyingCow ***@***.***>
wrote:

> Hi, already did. Do I need to wait for a couple hours for it to take
> effect? I can access Fulfillment Inbound API v0 just fine but not this
> latest API.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/amzn/selling-partner-api-models/discussions/3833#discussioncomment-10084817>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIBEMSZY4VM2JCGVLDL7FRLZM67K7AVCNFSM6AAAAABIWUYYYOVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAMBYGQ4DCNY>
> .
> You are receiving this because you commented.Message ID:
> <amzn/selling-partner-api-models/repo-discussions/3833/comments/10084817@
> github.com>
>


--

**Comment by itisdesign** [comment]
Amongst MWS's ListInboundShipments call's parameters were: LastUpdatedAfter and LastUpdatedBefore.

How do we achieve LastUpdated filtering w/ v2024-03-20's listInboundPlans call?

**Reply by itisdesign** [reply]
Related thread: https://github.com/amzn/selling-partner-api-models/discussions/574
